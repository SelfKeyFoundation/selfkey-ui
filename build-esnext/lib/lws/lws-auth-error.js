import * as React from 'react';
import { withStyles, createStyles } from '@material-ui/core';
import { LWSButton } from './lws-button';
import CommonStyle from '../common/common-style';
import { ShieldIcon } from '../icons/shield';
import { LWSError } from './lws-error';
const styles = createStyles({
    buttonPrimary: { ...CommonStyle.buttonPrimary, fontWeight: 700 },
    buttonSecondary: { ...CommonStyle.buttonSecondary, fontWeight: 700 }
});
export const LWSAuthError = withStyles(styles)(({ classes, website, retryAction }) => (React.createElement(LWSError, { actionIcon: React.createElement(ShieldIcon, null), actionName: "Authentication Error", errorName: `We can't connect you with ${website.name}`, supportText: "There\u2019s seems to be a problem with the authentication process at the moment. Please try again!", actionButton: React.createElement(LWSButton, { className: classes.buttonSecondary, onClick: retryAction }, "Retry") })));
export default LWSAuthError;
//# sourceMappingURL=lws-auth-error.js.map