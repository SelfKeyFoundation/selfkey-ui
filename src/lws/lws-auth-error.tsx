import * as React from 'react';
import injectSheet, { StyleSheet } from 'react-jss';
import { LWSButton } from './lws-button';
import CommonStyle from '../common/common-style';
import { ShieldIcon } from '../icons/shield';
import { LWSError } from './lws-error';
import { Website } from './lws-common';

const styles: StyleSheet = {
	buttonPrimary: CommonStyle.buttonPrimary,

	buttonSecondary: CommonStyle.buttonSecondary,
};

export type LWSAuthErrorProps = {
	website: Website;
	retryAction?: ((event: React.MouseEvent<HTMLElement>) => void);
};

export const LWSAuthError = injectSheet(styles)<LWSAuthErrorProps>(({ classes, website, retryAction }) => (
	<LWSError
		actionIcon={<ShieldIcon />}
		actionName="Authentication Error"
		errorName={`We can't connect you with ${website.name}`}
		supportText="Thereâ€™s seems to be a problem with the authentication process at the moment. Please try again!"
		actionButton={
			<LWSButton className={classes.buttonSecondary} onClick={retryAction}>
				Retry
			</LWSButton>
		}
	/>
));

export default LWSAuthError;
