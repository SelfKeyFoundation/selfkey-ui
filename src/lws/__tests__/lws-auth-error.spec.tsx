import * as React from 'react';
import { create } from 'react-test-renderer';
import { LWSAuthError } from '../lws-auth-error';

describe('LWS Auth Error Screen', () => {
	it('should show LWS Auth Error Screen', () => {
		const component = create(
			<LWSAuthError
				website={{
					name: 'PaveziCoin',
					url: 'http://www.pavezicoin.org',
					termsUrl: 'http://www.pavezicoin.org/terms',
					policyUrl: 'http://www.pavezicoin.org/privacy'
				}}
			/>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
