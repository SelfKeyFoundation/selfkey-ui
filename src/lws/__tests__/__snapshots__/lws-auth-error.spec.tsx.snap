// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LWS Auth Error Screen should show LWS Auth Error Screen 1`] = `
<div
  className="Component-lwsError-0-1-4"
>
  <div
    className="Component-areaTitle-0-1-7"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-1 Component-root-0-1-8"
      color={undefined}
      focusable="false"
      role="presentation"
      viewBox="0 0 68 74"
    >
      <title>
        icon-shield
      </title>
      <desc>
        Created with Sketch.
      </desc>
      <defs>
        <linearGradient
          id="linearGradient-1"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stopColor="#DB7400"
          />
          <stop
            offset="100%"
            stopColor="#B44000"
          />
        </linearGradient>
      </defs>
      <g
        fill="none"
        fillRule="evenodd"
        id="Modal---Dark"
        stroke="none"
        strokeWidth="1"
      >
        <g
          fill="url(#linearGradient-1)"
          fillRule="nonzero"
          id="2.1-Error---No-Selfkey-id"
          transform="translate(-686.000000, -277.000000)"
        >
          <g
            id="Modal-content"
            transform="translate(420.000000, 100.000000)"
          >
            <g
              id="icon-shield"
              transform="translate(266.000000, 177.000000)"
            >
              <path
                d="M33.9076923,0.0322121231 L33.3824516,0.292548369 C33.3824516,0.292548369 19.5846154,7.18461538 1.16923077,7.18461538 L0,7.18461538 L0,8.35384615 C0,35.4836538 8.28966375,51.8848561 16.6615385,61.4442308 C25.0288462,71.0081737 33.5468747,73.7759615 33.5468747,73.7759615 L33.9076923,73.8901439 L34.2685099,73.7759615 C34.2685099,73.7759615 42.7865385,71.0081737 51.1538462,61.4442308 C59.5257209,51.8848561 67.8153846,35.4836538 67.8153846,8.35384615 L67.8153846,7.18461538 L66.6461538,7.18461538 C48.2307692,7.18461538 34.432933,0.292548369 34.432933,0.292548369 L33.9076923,0.0322121231 Z M33.9076923,2.61730769 C35.2504814,3.26586597 48.0343747,9.18509674 65.4358176,9.46826923 C65.2028846,35.4197121 57.2831737,50.8983176 49.4,59.9096154 C41.6858176,68.7199516 34.4466352,71.195433 33.9076923,71.3781253 C33.3687506,71.195433 26.129567,68.7199516 18.4153846,59.9096154 C10.5322121,50.8983176 2.6125,35.4197121 2.37956702,9.46826923 C19.7810099,9.18509674 32.5649044,3.26586597 33.9076923,2.61730769 Z M31.7153846,8.65072086 L31.3088945,8.72379778 L31.1992791,8.76947145 L30.4867791,9.53677914 L30.6786055,10.5644231 L31.6194714,11.0348561 L32.0259615,10.9572121 L32.1355769,10.9161055 L32.8435099,10.1442308 L32.6516824,9.11658683 L31.7153846,8.65072086 Z M36.4836538,8.7375 L35.4423077,8.8608176 L34.917067,9.76971212 L35.3326923,10.7288462 L35.6889429,10.9343747 L35.7985583,10.9754814 L36.8399044,10.8521638 L37.3605769,9.94326923 L36.9495198,8.97956702 L36.5932692,8.77403846 L36.4836538,8.7375 Z M27.2576923,10.1990385 L26.8512022,10.2492791 L26.7415868,10.2812506 L25.9788462,11.0028846 L26.1021638,12.0442308 L27.0064901,12.5694714 L27.4175484,12.5192308 L27.5317308,12.4872593 L28.2899044,11.7656253 L28.1665868,10.7242791 L27.2576923,10.1990385 Z M40.9504814,10.2492791 L39.9182692,10.4365385 L39.4524044,11.3728362 L39.9228362,12.3091352 L40.2882209,12.4918275 L40.4024044,12.5237978 L41.4346154,12.3365385 L41.9050484,11.4048077 L41.4346154,10.4685099 L41.0646638,10.2812506 L40.9504814,10.2492791 Z M22.7269231,11.5007209 L22.3112978,11.5189901 L22.1971154,11.5509615 L21.3932692,12.2223561 L21.4480769,13.2682692 L22.325,13.8483176 L22.7314901,13.8254814 L22.8502407,13.7980769 L23.6540868,13.1266824 L23.5947121,12.0807692 L22.7269231,11.5007209 Z M45.5269231,11.5189901 L44.5038462,11.7382209 L44.0653846,12.692789 L44.567789,13.6108176 L44.9423077,13.7798077 L45.0564901,13.8117791 L46.079567,13.5925484 L46.5134615,12.6379814 L46.0156253,11.7199516 L45.6365385,11.5509615 L45.5269231,11.5189901 Z M18.0865385,12.5375 L17.6800484,12.5466352 L17.5612978,12.5694714 L16.7346154,13.2134615 L16.7528846,14.2593747 L17.6069714,14.8713945 L18.0180286,14.8622593 L18.1367791,14.8394231 L18.9634615,14.2 L18.9406253,13.1495198 L18.0865385,12.5375 Z M50.1125,12.5512022 L49.1076923,12.8480769 L48.7423077,13.8300484 L49.3086538,14.7115385 L49.6968747,14.8531253 L49.8156253,14.8759615 L50.820433,14.5790868 L51.1812506,13.5971154 L50.6103362,12.7156253 L50.2266824,12.5740385 L50.1125,12.5512022 Z M54.7848561,13.3276439 L53.7937506,13.6701923 L53.4740385,14.665866 L54.0769231,15.5199516 L54.4697121,15.6432692 L54.5884615,15.6615385 L55.575,15.3235583 L55.8992791,14.3278846 L55.2918275,13.4692308 L54.8990385,13.3459132 L54.7848561,13.3276439 Z M13.0031253,13.3504814 L12.8843747,13.3687506 L12.0394231,13.9807692 L12.0165868,15.0266824 L12.8432692,15.6706737 L13.2588945,15.6798077 L13.3730769,15.6615385 L14.2225967,15.0449516 L14.240866,13.9990385 L13.4141824,13.3596154 L13.0031253,13.3504814 Z M8.44038462,13.8848561 C8.41298135,13.880289 8.38557692,13.8848561 8.36274068,13.8848561 C8.29423077,13.8894231 8.27596154,13.8894231 8.30336597,13.8894231 L7.20264394,13.957933 L7.20264394,15.1088945 C7.19350991,15.5930286 7.4858176,16.0314901 7.93798135,16.2096154 C8.38557692,16.3877407 8.9016824,16.2735583 9.23052855,15.9173077 C9.58677914,15.5930286 9.71009674,15.0814901 9.53653846,14.6293275 C9.35841317,14.1771638 8.92451982,13.880289 8.44038462,13.8848561 Z M59.3384615,13.8848561 C58.8543275,13.8939901 58.425,14.2091352 58.2651439,14.665866 C58.105289,15.1271638 58.2468747,15.6341352 58.6213945,15.9492791 C58.6213945,15.9492791 58.6213945,15.9492791 58.6259615,15.9538462 C58.9593747,16.2872593 59.4663462,16.3831737 59.9048077,16.1959132 C60.3387022,16.0086538 60.6218747,15.5793275 60.6173077,15.1043275 L60.6173077,13.957933 L59.5165868,13.8894231 C59.5439901,13.8894231 59.5211538,13.8894231 59.4526439,13.8848561 C59.4161055,13.880289 59.379567,13.880289 59.3384615,13.8848561 Z M59.0507209,18.7262022 L58.2286055,19.3747593 L58.1235583,19.7766824 L58.1144231,19.895433 L58.5026439,20.8682692 L59.5165868,21.1377407 L60.3387022,20.4846154 L60.4437506,20.0872593 L60.4528846,19.9685099 L60.0646638,18.9956737 L59.0507209,18.7262022 Z M8.12980769,18.8221154 L7.39903846,19.5665868 L7.34423077,19.9730769 L7.34423077,20.0598561 L7.34879778,20.132933 L7.88774068,21.0326923 L8.92908683,21.1423077 L9.64158683,20.370433 L9.68269231,19.9593747 L9.68269231,19.9228362 L9.67812529,19.8862978 L9.67812529,19.8771638 L9.17115385,18.9591352 L8.12980769,18.8221154 Z M58.6305286,23.5766824 L57.7673077,24.170433 L57.6348561,24.5586538 L57.6165868,24.6774044 L57.945433,25.6730769 L58.9365385,26.0110583 L59.7951923,25.4173077 L59.9276439,25.0290868 L59.9459132,24.9057692 L59.6262022,23.9146638 L58.6305286,23.5766824 Z M8.46778905,23.7548077 L7.80552855,24.5632209 L7.78725932,24.9788462 L7.80096154,25.0975967 L8.39014394,25.9653846 L9.44062529,26.0064901 L10.1028846,25.1980769 L10.1211538,24.7870198 L10.1028846,24.6682692 L9.51370222,23.8004814 L8.46778905,23.7548077 Z M57.8586538,28.3769231 L56.9588945,28.915866 L56.7990385,29.2949516 L56.7762022,29.4137022 L57.0319714,30.4276439 L58.0002407,30.829567 L58.9,30.2951923 L59.0598561,29.9115385 L59.0826923,29.7973561 L58.8269231,28.7788462 L57.8586538,28.3769231 Z M10.2125,28.6235583 L9.16658683,28.6463945 L8.55456702,29.4913462 L8.56370222,29.9069714 L8.58653846,30.0211538 L9.22596154,30.8524044 L10.2764429,30.829567 L10.8884615,29.9800484 L10.8793275,29.5689901 L10.8564901,29.4502407 L10.2125,28.6235583 Z M56.7259615,33.1086538 L55.7896638,33.5836538 L55.6069714,33.9536055 L55.575,34.0723561 L55.7622593,35.1 L56.7031253,35.565866 L57.6394231,35.090866 L57.8221154,34.7209132 L57.8540868,34.6067308 L57.6622593,33.5790868 L56.7259615,33.1086538 Z M11.2538462,33.3826923 L10.2125,33.4694714 L9.65528905,34.3555286 L9.68725932,34.7665868 L9.71923077,34.8807692 L10.4134615,35.6663462 L11.4593747,35.579567 L12.0165868,34.6935099 L11.9800484,34.2824516 L11.9480769,34.1682692 L11.2538462,33.3826923 Z M55.2507209,37.7399044 L54.2778846,38.1189901 L54.0586538,38.4706737 L54.0175484,38.580289 L54.0997593,39.6262022 L54.9903846,40.1788462 L55.967789,39.7951923 L56.1870198,39.4480769 L56.2326923,39.3384615 L56.1413462,38.2925484 L55.2507209,37.7399044 Z M12.6377407,38.0276439 L11.6055286,38.2149044 L11.1396638,39.1557692 L11.2127407,39.5576923 L11.2538462,39.6718747 L12.0211538,40.3843747 L13.053366,40.1971154 L13.5192308,39.2608176 L13.4461538,38.8543275 L13.4050484,38.7401439 L12.6377407,38.0276439 Z M53.3461538,42.225 L52.3550484,42.5538462 L52.1175484,42.8963945 L52.0673077,43.0060099 L52.1084132,44.0519231 L52.967067,44.6456737 L53.9627407,44.3122593 L54.1956737,43.9742791 L54.2459132,43.8646638 L54.2048077,42.8141824 L53.3461538,42.225 Z M14.4555286,42.5355769 L13.4415868,42.8050484 L13.0487978,43.7778846 L13.1538462,44.1752407 L13.2086538,44.2848561 L14.0307692,44.9334132 L15.0401439,44.6639429 L15.432933,43.6911055 L15.3278846,43.2937506 L15.2730769,43.1887022 L14.4555286,42.5355769 Z M51.1401439,46.5365385 L50.1125,46.7512022 L49.8384615,47.0572121 L49.7790868,47.1576923 L49.6923077,48.2036055 L50.4733176,48.8978362 L51.5009615,48.6877407 L51.775,48.3771638 L51.8343747,48.2766824 L51.9211538,47.2307692 L51.1401439,46.5365385 Z M16.5884615,46.8699516 L15.6156253,47.2627407 L15.3461538,48.2766824 L15.5060099,48.6557692 L15.5653846,48.7608176 L16.4651439,49.2997593 L17.4334132,48.9115385 L17.7028846,47.8930286 L17.5430286,47.5139429 L17.4836538,47.4134615 L16.5884615,46.8699516 Z M48.5230769,50.6425484 L47.4862978,50.7932692 L47.1985583,51.0855769 L47.1300484,51.1860583 L46.9838945,52.2228362 L47.7237978,52.9627407 L48.7605769,52.8074516 L49.0528846,52.5151439 L49.1213945,52.4192308 L49.2675484,51.3824516 L48.5230769,50.6425484 Z M19.2375,50.9576923 L18.292067,51.4098561 L18.0865385,52.4375 L18.2692308,52.8074516 L18.3377407,52.907933 L19.2649044,53.392067 L20.2057692,52.9444714 L20.4112978,51.9122593 L20.2377407,51.5423077 L20.1646638,51.4463945 L19.2375,50.9576923 Z M45.6,54.5156253 L44.5540868,54.5567308 L44.2343747,54.8125 L44.1567308,54.9038462 L43.9009615,55.917789 L44.5586538,56.7307692 L45.604567,56.6942308 L45.9242791,56.4338945 L46.0019231,56.3425484 L46.2576923,55.3240385 L45.6,54.5156253 Z M22.2108176,54.7668275 L21.3247593,55.3240385 L21.2425484,56.3699516 L21.4617791,56.717067 L21.5394231,56.8038462 L22.5213945,57.182933 L23.4028846,56.6257209 L23.4896638,55.5798077 L23.270433,55.2326923 L23.1882209,55.1413462 L22.2108176,54.7668275 Z M41.2930286,58.0461538 L40.9459132,58.2699516 L40.8637022,58.3521638 L40.5028846,59.3387022 L41.078366,60.2110583 L42.1242791,60.275 L42.4713945,60.0512022 L42.5536055,59.9689901 L42.9144231,58.9824516 L42.3389429,58.1055286 L41.2930286,58.0461538 Z M25.5814901,58.2425484 L24.7548077,58.8911055 L24.7867791,59.9415868 L25.0379814,60.2612978 L25.1293275,60.3435099 L26.1432692,60.6129814 L26.9653846,59.9644231 L26.9379814,58.9185099 L26.6822121,58.5942308 L26.595433,58.5165868 L25.5814901,58.2425484 Z M37.6985583,61.1930286 L37.3331737,61.3848561 L37.2372593,61.457933 L36.7896638,62.403366 L37.2737978,63.3305286 L38.3105769,63.4949516 L38.6759615,63.3076923 L38.7673077,63.2346154 L39.2194714,62.2846154 L38.7307692,61.3574516 L37.6985583,61.1930286 Z M29.3175484,61.3300484 L28.5593747,62.0516824 L28.6872593,63.0930286 L28.970433,63.3944714 L29.0663462,63.4629814 L30.1031253,63.6365385 L30.8567308,62.9103362 L30.7334132,61.8735583 L30.4502407,61.5721154 L30.3497593,61.5036055 L29.3175484,61.3300484 Z M33.8163462,63.828366 C33.2728362,63.8375 32.8115385,64.2211538 32.6973561,64.7463945 C32.5831737,65.2762022 32.8435099,65.8151439 33.3276439,66.0480769 C33.3276439,66.0480769 33.3413462,66.0572121 33.3778846,66.0754814 L33.9031253,66.3358176 L34.4740385,66.0480769 C34.9673077,65.8105769 35.2276439,65.257933 35.0951923,64.7235583 C34.9673077,64.1891824 34.4786055,63.8146638 33.9305286,63.832933 C33.9259615,63.828366 33.9259615,63.828366 33.9213945,63.832933 C33.8848561,63.828366 33.8483176,63.828366 33.8163462,63.828366 Z M36.0322581,25.483871 C35.4731155,24.9247284 34.7957029,24.6451613 34,24.6451613 C33.2042971,24.6451613 32.5268845,24.9247284 31.9677419,25.483871 C31.4085994,26.0430135 31.1290323,26.7204261 31.1290323,27.516129 C31.1290323,28.3118319 31.4085994,28.9892445 31.9677419,29.5483871 C32.5268845,30.1075297 33.2042971,30.3870968 34,30.3870968 C34.7957029,30.3870968 35.4731155,30.1075297 36.0322581,29.5483871 C36.5914006,28.9892445 36.8709677,28.3118319 36.8709677,27.516129 C36.8709677,26.7204261 36.5914006,26.0430135 36.0322581,25.483871 Z M39.6451613,38.1290323 C37.7956897,37.3978458 35.9247406,37.0322581 34.0322581,37.0322581 C32.1397755,37.0322581 30.2688265,37.3870932 28.4193548,38.0967742 C26.5698832,38.8064552 25.6451613,39.4193523 25.6451613,39.9354839 L25.6451613,41.4193548 L42.4193548,41.4193548 L42.4193548,39.9354839 C42.4193548,39.4193523 41.4946329,38.8172077 39.6451613,38.1290323 Z M30.0967742,23.6129032 C31.1720484,22.537629 32.4731106,22 34,22 C35.5268894,22 36.8279516,22.537629 37.9032258,23.6129032 C38.9785,24.6881774 39.516129,25.9892397 39.516129,27.516129 C39.516129,29.0430184 38.9785,30.3440806 37.9032258,31.4193548 C36.8279516,32.494629 35.5268894,33.0322581 34,33.0322581 C32.4731106,33.0322581 31.1720484,32.494629 30.0967742,31.4193548 C29.0215,30.3440806 28.483871,29.0430184 28.483871,27.516129 C28.483871,25.9892397 29.0215,24.6881774 30.0967742,23.6129032 Z M29.516129,34.9677419 C31.1505458,34.5806432 32.6451545,34.3870968 34,34.3870968 C35.3548455,34.3870968 36.8494542,34.5806432 38.483871,34.9677419 C40.1182877,35.3548406 41.6128965,35.9892429 42.9677419,36.8709677 C44.3225874,37.7526926 45,38.7741877 45,39.9354839 L45,44.0645161 L23,44.0645161 L23,39.9354839 C23,38.7741877 23.6774126,37.7526926 25.0322581,36.8709677 C26.3871035,35.9892429 27.8817123,35.3548406 29.516129,34.9677419 Z"
                id="icon-warning-id"
              />
            </g>
          </g>
        </g>
      </g>
    </svg>
    <h2>
      Authentication Error
    </h2>
    <h3>
      We can't connect you with PaveziCoin
    </h3>
  </div>
  <div
    className="Component-form-0-1-3"
  >
    <p
      className="Component-supportText-0-1-6"
    >
      There’s seems to be a problem with the authentication process at the moment. Please try again!
    </p>
    <div
      className="Component-formSubmitRow-0-1-5"
    >
      <button
        className="Component-root-0-1-9 Component-buttonSecondary-0-1-2"
        onClick={undefined}
      >
        Retry
      </button>
    </div>
  </div>
</div>
`;
